Meeting notes for 2026-06-17 (originally 2026-06-03) (#134)
swicg/activitypub-trust-and-safety: meetings/2026-06-17-taskforce-meeting.md

- **Moderator-applied vs self labeling.** Labels need to distinguish first-party self-labeling from moderator-applied labeling via `attributedTo`. Where the server is also the actor's moderator (the common case), this collapses into `attributedTo` pointing at the moderation actor (the actor's `moderatedBy`), avoiding the need to subscribe to a separate moderation actor. a observed this closely resembles the annotation model without using it, and generalises to any case where a service publishes on a user's behalf and controls the canonical representation.
- **Upstream `Update` problem.** Server-to-server `Update` is a full replace, so the whole object must be re-sent (omitted fields are dropped); the C2S-partial vs S2S-full discrepancy is broken upstream and likely needs a Working Group fix. The report will have to work around it for now. a will propose in review that the label _application_ could itself be an `Activity` that flows through moderation actors/labelers.

> - **Default vocabulary.** Echo relayed internal review (Claire, David: looks good) and raised the lack of a default label vocabulary — risking a dozen duplicate definitions, aliasing maintenance, and inconsistent moderator handling. Emelia deliberately avoided defining a vocabulary in the report (bikeshedding risk; no clear place to host it at CG/WG level, though w3id.org exists) and suggested implementers — possibly with European.social — define a common one collaboratively. Echo noted Imani has already drafted a list intended for Mastodon, but the team wants to avoid Mastodon's choices becoming a de facto standard. Emelia noted European.social may be defunct (funding); Echo will check, as they may want to partner.

Reviewer: minor correction- we were talking about IFTAS and lack of funding there, not European.social. also talking about funding may not be terribly relevant to the minutes.

Author: I'd mentioned european.social since there's likely organisations involved in that (not the at protocol ones) that could help define a common vocabulary; IFTAS I wasn't sure of as being an option for this given how they've slowed down due to funding. Let me see if I can clarify:

```suggestion
- **Default vocabulary.** Echo relayed internal review (Claire, David: looks good) and raised the lack of a default label vocabulary — risking a dozen duplicate definitions, aliasing maintenance, and inconsistent moderator handling. Emelia deliberately avoided defining a vocabulary in the report (bikeshedding risk; no clear place to host it at CG/WG level, though w3id.org exists) and suggested implementers work together to define a common vocabulary — possibly with IFTAS or European.social partners. Echo noted Imani has already drafted a list intended for Mastodon, but the team wants to avoid Mastodon's choices becoming a de facto standard. Emelia noted IFTAS may not be able to work on this due to funding; Echo will check, as they may want to partner.
```

- **Emelia's pushback.** Introducing "host" is out of scope: it drags in ActivityPub's broken same-origin assumption and invites bikeshedding over what "host"/"server" means. ActivityPub has no real concept of a server. Better to lift that vocabulary into ActivityPub itself and keep the TF focused on a `moderationActor` and a `moderatedBy` property — which is more likely to be adopted by implementations and to bubble up to the Working Group.
- **a's argument.** The application actor is not necessarily the administrator (e.g. mastodon.social's admin is a designated _account_, not the application actor). a reworked the PR to borrow XMPP's "contact addresses" concept (explicit abuse/support contacts). The obstacle: XMPP has an explicit notion of the service; ActivityPub doesn't, so the concept exists in the Fediverse but can't be reasoned about because the vocabulary is missing. There is a FEP on identifying the application actor exposed via the NodeInfo link ([FEP-2677](https://fediverse.codeberg.page/fep/fep/2677/)), plus a related FEP Emelia couldn't name on the call — [FEP-f1d5](https://fediverse.codeberg.page/fep/fep/f1d5/), which formalises ActivityPub + NodeInfo.

> - **Hats.** Wearing his WG hat, a agreed this will likely land in ActivityPub eventually; wearing his CG/TF hat, he asked how useful the report's background framing is now — it does usefully capture that there are multiple layers and that you can't trust a self-asserted link because anyone can lie about who moderates them.

Reviewer 2: s/his/their

Author: ```suggestion
- **Hats.** Wearing their WG hat, a agreed this will likely land in ActivityPub eventually; wearing their CG/TF hat, they asked how useful the report's background framing is now — it does usefully capture that there are multiple layers and that you can't trust a self-asserted link because anyone can lie about who moderates them.
```

My apologies @trwnh I had AI draft these and was too fatigued last night to catch everything.

**Outcome:** Task Force stays scoped to `moderatedBy` plus a moderation inbox; the "host" concept is deferred to ActivityPub itself. Verifying the `moderatedBy` relationship is the central threat to solve (see below).

### Verifying the `moderatedBy` relationship (threat modeling)

> The bulk of the meeting. The core threat: a malicious actor (e.g. `evil.com`) falsely declaring `moderatedBy: mastodon.social`. The system needs to verify that the named moderation actor has actually consented to moderating that actor. Echo agreed this is a real threat to model.

Reviewer: this conversation was not the bulk of the meeting, if i recall it happened after the hour mark

Author: Yeah, there was the hour mark where we'd covered item 1 for 15 minutes, item 2 for 30 and then had 15 minutes where we started on moderatedBy, but then went overtime by 30-45 minutes on that. So that's how it became the bulk of the meeting.

Reviewer: also looking at this again, i'm not sure i agreed this was a real threat. quite the contrary, i still don't buy this is a real problem.

Author: Why not? I, a non-mastodon.social actor could advertise mastodon.social's moderation actor as mine, and then that would cause you to receive all my Flag'd activity; others would have no idea you don't moderate me, and with C2S+Handles we cannot simply limit to same origin

Reviewer: i'm going to comment on the issue or PR later today and we can discuss it there. for now i'm just saying i don't recall agreeing it was a real threat in the meeting, i just wanted to move forward in the discussion
